src/index.ts

```diff
@@ -38,7 +38,11 @@ function yn (value: string | undefined) {
  * Debugging `ts-node`.
  */
 const shouldDebug = yn(process.env.TS_NODE_DEBUG)
-const debug = shouldDebug ? console.log.bind(console, 'ts-node') : () => undefined
+const timestamp = function(){};
+timestamp.toString = function(){
+  return "[ts-node " + (new Date).toISOString() + "]";
+}
+const debug = shouldDebug ? console.log.bind(console, '%s', timestamp) : () => undefined
```

Comment:
IMO we should simplifying by writing a function.

`function(...args) {console.log(timestamp(), ...args)}`